### PR TITLE
Add ratings, reviews and icons into Marketplace's Product Cards

### PR DESCRIPTION
**Pull request by tgglv**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The search on _WooCommerce > Marketplace_ page that shows product from the WooCommerce.com marketplace as cards. During previous re-design phase we added updated product cards in https://github.com/woocommerce/woocommerce/pull/30410.

This PR updates those product cards by:
- replacing a product image with icon (if it exists) that fits better
- mentioning a vendor who developed a product together with a link to reach the rest of their products on WooCommerce.com
- adding an average rating represented as 5 stars (golden, gray, or 50/50)
- adding the number of reviews left by customers for the product

Also, the text color for recurring payment cadence ("per year") was changed according to [the design](seb2o1DExPxoh5d53ftJif-fi-812%3A4611).

Closes 10867-gh-Automattic/woocommerce.com

### How to test the changes in this Pull Request:

1. Check out this branch to your local environment
2. Go to _WooCommerce > Marketplace_
3. Search for "memberships"
4. Ensure you see products like _WooCommerce Memberships_ that contain:
    - icon (instead of a product image), 
    - developed by ... (with a link to a vendor page on WooCommerce.com)
    - average rating represented as stars
    - number of reviews left by customers

Screen Shot 2021-09-30 at 11 07 48

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enhancement - Add ratings, reviews and icons into Marketplace's Product Cards.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by tgglv** [comment]
Local test run results:
```
Test Suites: 2 failed, 2 skipped, 36 passed, 38 of 40 total
Tests:       9 failed, 5 skipped, 173 passed, 187 total
```

Failed scenarios:
- Import Products from a CSV file › can upload the CSV file and import products
- A japanese store can complete the selective bundle install but does not include WCPay. › should display the choose payments task, and not the woocommerce payments task

It looks like failed tests aren't connected with the changes.

File: includes/admin/views/html-admin-page-addons.php
Comment (andfinally) [inline includes/admin/views/html-admin-page-addons.php]: It might be good to add `target="_blank"` for this one – unless you think it's not a good experience.

File: includes/admin/views/html-admin-page-addons.php
Comment (andfinally) [inline includes/admin/views/html-admin-page-addons.php]: I'm thinking we shouldn't show the image here if there's no icon: it's not designed for this scale, and won't always look good; and it'll hopefully encourage extension developers to add an app icon to their extensions.

**Review by andfinally — DISMISSED** [review]
Looks great, thanks! I just left a couple of minor suggestions.

File: includes/admin/views/html-admin-page-addons.php
Comment (tgglv) [reply includes/admin/views/html-admin-page-addons.php]: That's a great suggestion! Thanks, @andfinally! Applied in https://github.com/woocommerce/woocommerce/pull/30840/commits/2461d20e2d51482a2ed83c3b97d9a44f8c0479cb.

File: includes/admin/views/html-admin-page-addons.php
Comment (tgglv) [reply includes/admin/views/html-admin-page-addons.php]: Indeed, this way we'd encourage vendors to use icons. Applied in https://github.com/woocommerce/woocommerce/pull/30840/commits/2461d20e2d51482a2ed83c3b97d9a44f8c0479cb.

File: includes/admin/views/html-admin-page-addons.php
Comment (kloon) [inline includes/admin/views/html-admin-page-addons.php]: Can we add utm params to these links?

File: includes/admin/views/html-admin-page-addons.php
Comment (kloon) [inline includes/admin/views/html-admin-page-addons.php]: How about we fetch the suffix from the API with a fallback to per year?

**Review by kloon — CHANGES_REQUESTED** [review]
Changes looks great so far @tgglv, just left a small couple of comments that needs addressing.

**Comment by corsonr** [comment]
@tgglv I just tested it, and that works beautifully! Just a small comment, I'd suggest that the comments in the `admin.scss` file use the following format `// comment` instead of `/* comment */` so that they do not render in the compiled file.

I made the changes in this [diff file](https://d.pr/f/gVvVJN+) in case you want to implement it.

**Review by andfinally — DISMISSED** [review]
Thanks for the changes, looking good! Approving on the assumption we'll apply Gerhard and Remi's suggestions.

**Comment by andfinally** [comment]
I took the liberty to pushing Remi's comment changes to the branch. Still waiting for confirmation on the UTM tags from marketing.

File: includes/admin/views/html-admin-page-addons.php
Comment (vedanshujain) [inline includes/admin/views/html-admin-page-addons.php]: can we move this function to `class-wc-admin-addons.php`? seems like this can be used in multiple views in the future and not just this one.

File: includes/admin/views/html-admin-page-addons.php
Comment (andfinally) [reply includes/admin/views/html-admin-page-addons.php]: Marketing would like us to use these ones:

```
utm_source=extensionsscreen
utm_medium=product
utm_campaign=wcaddons
utm_content=devpartner
```

**Review by vedanshujain — COMMENTED** [review]
Left a minor comment. Looks like tests are also failing but seems unrelated, can you please rebase with latest trunk to see if that fixes the failing tests.

File: includes/admin/views/html-admin-page-addons.php
Comment (andfinally) [reply includes/admin/views/html-admin-page-addons.php]: Added in d0eeea24e4310dc31ddc66f990d8c5167cd9e9db.

File: includes/admin/views/html-admin-page-addons.php
Comment (andfinally) [reply includes/admin/views/html-admin-page-addons.php]: Added in 4e370a4f94afad7d82a9c527c8ace77efe835f61. I'm not sure how we'd translate such strings – we'll have to start passing the locale to the endpoint. Added 11488-gh-Automattic/woocommerce.com for the back end changes.

File: includes/admin/views/html-admin-page-addons.php
Comment (tgglv) [reply includes/admin/views/html-admin-page-addons.php]: Thanks, @vedanshujain! Addressed in https://github.com/woocommerce/woocommerce/pull/30840/commits/8e2f8bd1db3dc28e7c2111009cc9f8ebed11565a.

**Comment by tgglv** [comment]
> Left a minor comment. Looks like tests are also failing but seems unrelated, can you please rebase with latest trunk to see if that fixes the failing tests.

@vedanshujain, it seems that the problem is in the `trunk` as well.
Solved the problem with tests by changing hash in the `composer.lock`. It looks like coincidentally phpunit files in cache break running `composer install`.

**Review by vedanshujain — DISMISSED** [review]
Thanks for the changes, LGTM! I am not merging it just yet, and will leave it to @Konamiman to see where we want to fix the composer for unit tests.

**Comment by Konamiman** [comment]
> I am not merging it just yet, and will leave it to @Konamiman to see where we want to fix the composer for unit tests.

Also there are merge conflicts, once fixed let's see what can we do with that `composer.json` thing.

**Pull request merged**
